test.js

```diff
@@ -407,6 +407,35 @@ test('escapePath', t => {
 	});
 });
 
+test('deepKeys', t => {
+	t.deepEqual(deepKeys({
+		'a.b': {
+			c: {
+				d: [1, 2, 3],
+				e: '🦄',
+				f: 0,
+			},
+			'': {
+				a: 0,
+			},
+		},
+		'': {
+			a: 0,
+		},
+	}), [
+		'a\\.b.c.d[0]',
+		'a\\.b.c.d[1]',
+		'a\\.b.c.d[2]',
+		'a\\.b.c.e',
+		'a\\.b.c.f',
+		'a\\.b..a',
+		'.a',
+	]);
```

Comment:
It would be good with a test that take the output here and tries to use each path, to ensure that they are all valid.